src/overlays/actors/ovl_En_Wf/z_en_wf.c

```diff
@@ -137,81 +144,1344 @@ static ColliderCylinderInit D_80B37A5C = {
+            EffectSsDeadDb_Spawn(globalCtx, &pos, &velAndAccel, &velAndAccel, 100, 0, 255, 255, 255, 255, 0, 0, 255, 1,
+                                 9, true);
+        }
+    }
+}
+
+void EnWf_FaceTowardsPlayer(EnWf* this, GlobalContext* globalCtx) {
+    if ((this->action == WOLFOS_ACTION_WAIT) && (this->headTilt != 0)) {
+        this->headRot.y = Math_SinS(this->headTilt * 4200) * 8920.0f;
+    } else if (this->action != WOLFOS_ACTION_STUNNED) {
+        if (this->action != WOLFOS_ACTION_FIRST_SLASH) {
+            Math_SmoothStepToS(&this->headRot.y, this->actor.yawTowardsPlayer - this->actor.shape.rot.y, 1, 1500, 0);
+            this->headRot.y = CLAMP(this->headRot.y, -0x3127, 0x3127);
+        } else {
+            this->headRot.y = 0;
+        }
+    }
+}
+
+void EnWf_UpdateDamage(EnWf* this, GlobalContext* globalCtx) {
+    if (this->colliderSphere.base.acFlags & AC_BOUNCED) {
+        this->colliderSphere.base.acFlags &= ~(AC_HIT | AC_BOUNCED);
+        this->colliderCylinder1.base.acFlags &= ~AC_HIT;
+        this->colliderCylinder2.base.acFlags &= ~AC_HIT;
+    } else if ((this->colliderCylinder1.base.acFlags & AC_HIT) || (this->colliderCylinder2.base.acFlags & AC_HIT)) {
+        if (this->action >= WOLFOS_ACTION_WAIT) {
+            s16 yawDiff = this->actor.yawTowardsPlayer - this->actor.shape.rot.y;
+
+            if (((!(this->colliderCylinder1.base.acFlags & AC_HIT)) &&
+                 (this->colliderCylinder2.base.acFlags & AC_HIT)) ||
+                (ABS(yawDiff) > 19000)) {
+                this->actor.colChkInfo.damage *= 4;
+            }
+
+            this->colliderCylinder1.base.acFlags &= ~AC_HIT;
+            this->colliderCylinder2.base.acFlags &= ~AC_HIT;
+
+            if (this->actor.colChkInfo.damageEffect != ENWF_DMGEFF_ICE_MAGIC) {
+                this->damageEffect = this->actor.colChkInfo.damageEffect;
+                Actor_SetDropFlag(&this->actor, &this->colliderCylinder1.info, 1);
+                this->slashStatus = 0;
+
+                if ((this->actor.colChkInfo.damageEffect == ENWF_DMGEFF_STUN) ||
+                    (this->actor.colChkInfo.damageEffect == ENWF_DMGEFF_UNDEF)) {
+                    if (this->action != WOLFOS_ACTION_STUNNED) {
+                        Actor_SetColorFilter(&this->actor, 0, 120, 0, 80);
+                        Actor_ApplyDamage(&this->actor);
+                        EnWf_SetupStunned(this);
+                    }
+                } else { // EN_WF_DMGEFF_FIRE or NONE
+                    Actor_SetColorFilter(&this->actor, 0x4000, 255, 0, 8);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wf/EnWf_Destroy.s")
+                    if (this->damageEffect == ENWF_DMGEFF_FIRE) {
+                        this->fireTimer = 40;
+                    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wf/func_80B33FB0.s")
+                    if (!(Actor_ApplyDamage(&this->actor))) {
```

Comment:
```suggestion
                    if (!Actor_ApplyDamage(&this->actor)) {
```

Reply:
`== 0`, I hope, since this doesn't return a bool.